mypy/checker.py

```diff
@@ -2813,14 +2813,24 @@ def infer_partial_type(self, name: Var, lvalue: Lvalue, init_type: Type) -> bool
             partial_type = PartialType(None, name)
         elif isinstance(init_type, Instance):
             fullname = init_type.type.fullname
-            if (isinstance(lvalue, (NameExpr, MemberExpr)) and
+            is_ref = isinstance(lvalue, (NameExpr, MemberExpr))
```

Comment:
While you are at it maybe change this to `RefExpr`?

Reply:
Makes sense.